arbnode/resourcemanager/resource_management_test.go

```diff
@@ -37,42 +36,57 @@ total_rss 1016209408
 total_inactive_file %d
 total_active_file 321544192
 `, inactive)
-	if err != nil {
-		return err
+	return err
+}
+
+func makeCgroupsTestDir(cgroupDir string) cgroupsMemoryFiles {
+	return cgroupsMemoryFiles{
+		limitFile:  cgroupDir + "/memory.limit_in_bytes",
+		usageFile:  cgroupDir + "/memory.usage_in_bytes",
+		statsFile:  cgroupDir + "/memory.stat",
+		inactiveRe: regexp.MustCompile(`total_inactive_file (\d+)`),
 	}
-	return nil
 }
 
-func TestCgroupsv1MemoryLimit(t *testing.T) {
-	cgroupDir := t.TempDir()
-	c := newCgroupsV1MemoryLimitChecker(cgroupDir, 95)
-	_, err := c.isLimitExceeded()
-	if err == nil {
-		t.Error("Should fail open if can't read files")
+func TestCgroupsFailIfCantOpen(t *testing.T) {
+	testFiles := makeCgroupsTestDir(t.TempDir())
+	c := newCgroupsMemoryLimitChecker(testFiles, 95)
+	var err error
+	if _, err = c.isLimitExceeded(); err == nil {
+		t.Fatal("Should fail open if can't read files")
 	}
+}
 
-	err = updateFakeCgroupv1Files(c, 1000, 1000, 51)
-	if err != nil {
-		t.Error(err)
+func TestCgroupsLimitNotExceeded(t *testing.T) {
+	testFiles := makeCgroupsTestDir(t.TempDir())
+	c := newCgroupsMemoryLimitChecker(testFiles, 95)
+
+	var err error
+	if err = updateFakeCgroupFiles(c, 1000, 1000, 51); err != nil {
+		t.Fatal(err)
 	}
 	exceeded, err := c.isLimitExceeded()
 	if err != nil {
-		t.Error(err)
+		t.Fatal(err)
 	}
 	if exceeded {
-		t.Error("Expected under limit")
+		t.Fatal("Expected under limit")
```

Comment:
Doesn't matter too much in this case, but once you change this to being table-driven tests, this should be Errorf.


https://google.github.io/styleguide/go/decisions#keep-going

Reply:
Thanks, that's interesting to know. I had always followed the opposite pattern (fail fast), but I'll try to use this where it makes sense.